Tests/Swift Tests/BugsnagSwiftPublicAPITests.swift

```diff
@@ -0,0 +1,368 @@
+ */
+
+class FakePlugin: NSObject, BugsnagPlugin {
+    func load(_ client: BugsnagClient!) {}
+    func unload() {}
+}
+
+// MetadataStore conformance - presence of required methods is a test in and of itself
+class myMetadata: NSObject, BugsnagMetadataStore, BugsnagClassLevelMetadataStore {
+    static func addMetadata(_ metadata: [AnyHashable : Any], section sectionName: String) {}
+    static func addMetadata(_ metadata: Any?, key: String, section sectionName: String) {}
+    static func getMetadata(section sectionName: String) -> NSMutableDictionary? { return NSMutableDictionary() }
+    static func getMetadata(section sectionName: String, key: String) -> Any? { return nil }
+    static func clearMetadata(section sectionName: String) {}
+    static func clearMetadata(section sectionName: String, key: String) {}
+    
+    func addMetadata(_ metadata: [AnyHashable : Any], section sectionName: String) {}
+    func addMetadata(_ metadata: Any?, key: String, section sectionName: String) {}
+    func getMetadata(section sectionName: String) -> NSMutableDictionary? { return NSMutableDictionary() }
+    func getMetadata(section sectionName: String, key: String) -> Any? { return nil }
+    func clearMetadata(section sectionName: String) {}
+    func clearMetadata(section sectionName: String, key: String) {}
+}
+
+class BugsnagSwiftPublicAPITests: XCTestCase {
+
+    let apiKey = "01234567890123456789012345678901"
+    let ex = NSException(name: NSExceptionName("exception"),
+                         reason: "myReason",
+                         userInfo: nil)
+    let err = NSError(domain: "dom", code: 123, userInfo: nil)
+    let sessionBlock: BugsnagOnSessionBlock = { (session) -> Bool in return false }
+    let onSendErrorBlock: BugsnagOnSendErrorBlock = { (event) -> Bool in return false }
+    let onBreadcrumbBlock: BugsnagOnBreadcrumbBlock = { (breadcrumb) -> Bool in return false }
+    
+    func testBugsnagClass() throws {
+        Bugsnag.start(withApiKey: "")
+        Bugsnag.start(withApiKey: apiKey);
+        Bugsnag.start(with: BugsnagConfiguration(apiKey))
+        
+        let _ = Bugsnag.appDidCrashLastLaunch()
+        
+        Bugsnag.notify(ex)
+        Bugsnag.notify(ex) { (event) -> Bool in return false }
+        Bugsnag.notifyError(err)
+        Bugsnag.notifyError(err) { (event) -> Bool in return false }
+        
+        Bugsnag.leaveBreadcrumb(withMessage: "msg")
+        Bugsnag.leaveBreadcrumb(forNotificationName: "notif")
+        Bugsnag.leaveBreadcrumb("msg", metadata: ["foo" : "bar"], type: .error)
+        
+        Bugsnag.startSession()
+        Bugsnag.pauseSession()
+        Bugsnag.resumeSession()
+        
+        Bugsnag.setContext("ctx")
+        let _ = Bugsnag.context()
+        
+        Bugsnag.setUser("me", withEmail: "memail@foo.com", andName: "you")
+        let _ = Bugsnag.user()
+        
+        Bugsnag.addOnSession(block: sessionBlock)
```

Comment:
would be good to verify that trailing block closures are supported too (ditto for config/client):

```
Bugsnag.addOnSession { (block) -> Bool in
    return true;
}
```